snap/hooks/configure

```diff
@@ -1,2 +1,20 @@
 #!/bin/sh
-snapctl restart $SNAP_NAME
+
+if [ "$(snapctl get daemon)" = "" ]
+then
+  # By default, we run as a daemon on core, otherwise configure the daemon to stop
+  if grep -q -e snap_core= -e snapd_recovery_mode= /proc/cmdline
+  then snapctl set daemon=true
+  else snapctl set daemon=false
+  fi
+fi
+
+if [ "$(snapctl get daemon)" = "true" ]; then
+  if snapctl services ${SNAP_NAME}.daemon | grep -q inactive; then
+    snapctl start --enable $SNAP_NAME.daemon 2>&1 || true
+  fi
+else
+  if snapctl services ${SNAP_NAME}.daemon | grep -q enabled; then
+    snapctl stop --disable $SNAP_NAME.daemon 2>&1 || true
+  fi
+fi
```

Comment:
The `if`s here can be dropped, both `start --enable` and `stop --disable` are idempotent?

```suggestion
if [ "$(snapctl get daemon)" = "true" ]; then
  snapctl start --enable $SNAP_NAME.daemon 2>&1 || true
else
  snapctl stop --disable $SNAP_NAME.daemon 2>&1 || true
fi
```